src/org/rascalmpl/library/String.rsc

```diff
@@ -524,8 +524,7 @@ public str squeeze(str src, str charSet) {
     return src;
   }
   return visit(src) {
-    case /<c:.><c>+/ => c
-      when /[<charSet>]/ := c
+    case /<c:[<charSet>]><c>+/ => c
```

Comment:
This is not valid Rascal code, although it may accidentally work in the Interpreter. We can only splice literals into regular expressions, not character classes or other meta symbols.

Reply:
Typically regular expressions are expected to be compiled at compile-time. That's why we have this limitation.

Reply:
yes, it indeed worked, I was a bit surprised as well. hmm, a pitty we don't allow backreferences in our regexes. In that case I should revive the java implementation I guess.